### PR TITLE
Update grpc java to 1.20

### LINKED ISSUE #189: Remove AuthenticatingClientInterceptors
The authentication via `ClientInterceptor`s is potentially less secure than `CallCredentials` and thus should be replaced everywhere.

(And also removes some `@Deprecated` code)

### LINKED ISSUE #191: Remove deprecated host and port client properties
The properties have been deprecated in 2.2.0 and should be removed.

(Or we could wait a little bit longer)

### LINKED ISSUE #203: Please upgrade to the recent version of gRPC-java dependencies
If possible, could you please upgrade the version of grpc-java? Version 1.1.19 fixes a couple of things, including some annoying warnings on JDK 11.

Thanks!

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #203 
Fixes #191

This PR updates grpc-java to 1.20 (was 1.18).

**Note:** This DROPs support for versions prior to 1.19 (for the client side).

Client and server side have been tested to work with 1.19 and 1.20

## Changes

* Update to grpc-java 1.20.0
* Use grpc-java bom for grpc dependency management
* Fixed `Bean '...' is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)` message
* Do not create `LoadBalancer.Factory` beans anymore, but add the `defaultLoadBalancingPolicy` client config. Custom `LoadBalancer.Factory`s will still be registered as before.
* Update to new `CallCredentials` classes
* Drop the deprecated `AuthenticatingClientInterceptors` support and simplify the tests accordingly
* Improve formatting and fix reported code-quality issues

----------------

@lexuna Thanks for your help with the update.

### CODE REVIEW

**Comment by ST-DDT** [comment]
@yidongnan Should I split the PR into multiple PRs (update gRPC, fix BeanPostProcessors message, remove `AuthenticatingClientInterceptors`, fix formatting)

**Comment by yidongnan** [comment]
It is certainly better to separate these PRs. If we don't separate, we can also use issues and commit message to track. 😄

**Comment by ST-DDT** [comment]
Splitted into #218, #219, #220 

Now this is only contains the actual update.

**Comment by ST-DDT** [comment]
All PRs are ready for review and merge.

File: tests/build.gradle [hunk 1]
```diff
@@ -73,7 +73,7 @@ dependencies {
         compile 'javax.annotation:javax.annotation-api:1.3.1'
     }
     testImplementation(
-        "io.grpc:grpc-testing:${grpcVersion}",
+        "io.grpc:grpc-testing",
```
[the remark this answers is not in the document]
Comment (ST-DDT) [reply tests/build.gradle]: Changed them both.

**Pull request merged**